tst/EnergyPlus/unit/SQLite.unit.cc

```diff
@@ -0,0 +1,1055 @@
+// EnergyPlus::SQLite Unit Tests
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include <EnergyPlus/SQLiteProcedures.hh>
+
+using namespace EnergyPlus;
+using namespace ObjexxFCL;
+
+namespace EnergyPlus {
+
+	class SQLiteFixture : public testing::Test
+	{
+
+	protected:
+		static void SetUpTestCase() { }
+
+		static void TearDownTestCase() { }
+
+		std::unique_ptr<SQLite> sqlite_test;
+		std::ostringstream ss;
+
+		virtual void SetUp() {
+			ASSERT_NO_THROW(sqlite_test = std::unique_ptr<SQLite>(new SQLite( ss, ":memory:", "std::ostringstream", true, true )));
+			ASSERT_TRUE(sqlite_test->writeOutputToSQLite());
+			ASSERT_TRUE(sqlite_test->writeTabularDataToSQLite());
+			sqlite_test->sqliteExecuteCommand("PRAGMA foreign_keys = ON;");
+			EXPECT_EQ("SQLite3 message, std::ostringstream open for processing!\n", ss.str());
+			ss.str(std::string());
+		}
+
+		virtual void TearDown() {
+			// don't know if this is needed...
+			sqlite_test.reset();
+			sqlite_test = nullptr;
+		}
+
+		std::string storageType( const int storageTypeIndex ) {
+			return sqlite_test->storageType( storageTypeIndex );
+		}
+
+		std::string timestepTypeName( const int timestepType ) {
+			return sqlite_test->timestepTypeName( timestepType );
+		}
+
+		std::string reportingFreqName( const int reportingFreqIndex ) {
+			return sqlite_test->reportingFreqName( reportingFreqIndex );
+		}
+
+		std::string columnText( const unsigned char* column ) {
+			return std::string( reinterpret_cast<const char*>( column ) );
```

Comment:
I actually copied this from [OpenStudio](https://github.com/NREL/OpenStudio/blob/23ab299e80c80c9d57a857e6300b3701340d2f76/openstudiocore/src/utilities/sql/SqlFile_Impl.cpp#L48) ...